tests/1_host_configuration.sh

```diff
@@ -2,474 +2,443 @@
   fi
 
   if mountpoint -q -- "$docker_root_dir" >/dev/null 2>&1; then
-    pass "$check_1_2_1"
-    resulttestjson "PASS"
-    currentScore=$((currentScore + 1))
+    pass -s "$check"
+    logcheckresult "PASS"
   else
-    warn "$check_1_2_1"
-    resulttestjson "WARN"
-    currentScore=$((currentScore - 1))
+    warn -s "$check"
+    logcheckresult "WARN"
   fi
 }
 
-# 1.2.2
-check_1_2_2() {
-  id_1_2_2="1.2.2"
-  desc_1_2_2="Ensure only trusted users are allowed to control Docker daemon (Scored)"
-  check_1_2_2="$id_1_2_2  - $desc_1_2_2"
-  starttestjson "$id_1_2_2" "$desc_1_2_2"
+check_1_2_2() { 
+  local id="1.2.2"
+  local desc="Ensure only trusted users are allowed to control Docker daemon (Scored)"
+  local remediation="You should remove any untrusted users from the docker group using command sudo gpasswd -d <your-user> docker or add trusted users to the docker group using command sudo usermod -aG docker <your-user>. You should not create a mapping of sensitive directories from the host to container volumes."
+  local remediationImpact="Only trust user are allow to build and execute containers as normal user."
+  local check="$id - $desc"
+  starttestjson "$id" "$desc"
 
-  totalChecks=$((totalChecks + 1))
   if command -v getent >/dev/null 2>&1; then
     docker_users=$(getent group docker)
   else
     docker_users=$(grep 'docker' /etc/group)
   fi
-  info "$check_1_2_2"
-  for u in $docker_users; do
-    info "       * $u"
-  done
-  resulttestjson "INFO" "users" "$docker_users"
-  currentScore=$((currentScore + 0))
+  docker_users=$(printf "%s" "$docker_users" | awk -F: '{print $4}')
+
+  local doubtfulusers=""
+  if [ -n "$dockertrustusers" ]; then
+    for u in $(printf "%s" "$docker_users" | sed "s/,/ /g"); do
+      if ! printf "%s" "$dockertrustusers" | grep -q "$u" ; then
+        if [ -n "${doubtfulusers}" ]; then
+          doubtfulusers="${doubtfulusers},$u"
+        else
+          doubtfulusers="$u"
+        fi
+      fi
+    done
+  else
+    info -c "$check"
+    info "      * Users: $docker_users"
+    logcheckresult "INFO" "doubtfulusers" "$docker_users"
+  fi
+
+  if [ -n "${doubtfulusers}" ]; then
+    warn -s "$check"
+    warn "      * Doubtful users: $doubtfulusers"
+    logcheckresult "WARN" "doubtfulusers" "$doubtfulusers"
+  fi
+
+  if [ -z "${doubtfulusers}" ] && [ -n "${dockertrustusers}" ]; then
+    pass -s "$check"
+    logcheckresult "PASS"
+  fi
 }
 
-# 1.2.3
 check_1_2_3() {
-  id_1_2_3="1.2.3"
-  desc_1_2_3="Ensure auditing is configured for the Docker daemon (Scored)"
-  check_1_2_3="$id_1_2_3  - $desc_1_2_3"
-  starttestjson "$id_1_2_3" "$desc_1_2_3"
+  local id="1.2.3"
+  local desc="Ensure auditing is configured for the Docker daemon (Scored)"
+  local remediation="Install and configure auditd using command sudo apt-get install auditd. Add -w /usr/bin/dockerd -k docker to the /etc/audit/rules.d/audit.rules file. Then restart the audit daemon using command service auditd restart."
```

Comment:
The package manager stuff needs to be generic, `apt-get` is Debian family for example